lib/chef/knife/vsphere_vm_clone.rb

```diff
@@ -233,13 +234,25 @@ class Chef::Knife::VsphereVmClone < Chef::Knife::BaseVsphereCommand
          description: 'Indicates whether to mark the new vm as a template',
          boolean: false
 
+  option :random_vmname,
+         long: '--random-vmname',
+         description: 'Creates a random VMNAME starts with vm-XXXXXXXX',
+         boolean: false
+
+  option :random_vmname_prefix,
+         long: '--random-vmname-prefix PREFIX',
+         description: 'Change the VMNAME prefix',
+         default: 'vm-'
+
   def run
     $stdout.sync = true
 
     vmname = @name_args[0]
-    if vmname.nil?
+    if (vmname.nil? && !config[:random_vmname]) || (vmname && config[:random_vmname])
```

Comment:
For sure. That `@random_hostname ||= randomstuff` memoizes it so that it returns the same thing each time it's called. If you think it makes more sense to memoize `vmname` itself instead, that's cool, too.